src/MPDevice.cpp

```diff
@@ -1750,8 +1820,21 @@ void MPDevice::detagPointedNodes(Common::AddressType addrType /*= Common::CRED_A
     }
 }
 
+void MPDevice::detagPointedNotesNodes()
+{
+    for (auto &i: notesLoginNodes)
+    {
+        i->removePointedToCheck();
+    }
+    for (auto &i: notesLoginChildNodes)
+    {
+        i->removePointedToCheck();
+    }
+}
+
 /* Follow the chain to tag pointed nodes (useful when doing integrity check when we are getting everything we can) */
-bool MPDevice::tagPointedNodes(bool tagCredentials, bool tagData, bool repairAllowed, Common::AddressType addrType /*= Common::CRED_ADDR_IDX*/)
+bool MPDevice::tagPointedNodes(bool tagCredentials, bool tagData, bool repairAllowed, Common::AddressType addrType /*= Common::CRED_ADDR_IDX*/,
+                               Common::DataAddressType dataAddrType /*= Common::DATA_ADDR_IDX*/)
```

Comment:
that may be a stupid question: have you checked that we never call tagPointedNodes with true, true? I mean, is there a use case where we need to tag all data types & credential types at once, as this code wouldn't be able to do it?

Reply:
actually it seems it happens in checkloadednodes in:

if (checkCredentials || checkData)
          {
              return_bool &amp;= tagPointedNodes(checkCredentials, checkData, repairAllowed);
          }

Reply:
actually you kind of do it after with return_bool &= tagPointedNodes(true, checkData, repairAllowed, Common::WEBAUTHN_ADDR_IDX);

this is getting a tad complex IMHO... should we change tagPointedNodes to not use Common::AddressType addrType & Common::DataAddressType dataAddrTyp and use two bools instead?